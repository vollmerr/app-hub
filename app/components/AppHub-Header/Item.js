import styled from 'styled-components';

import theme from 'utils/theme';

const focusStyle = (props) => `
  background: ${props.dark ? theme.blueDark : theme.themePrimary};
  border-bottom: 2px solid ${props.dark ? theme.orange : theme.orangeLighter};
`;

const Item = styled.div`
  display: flex;
  align-items: center;
  height: ${theme.hub.headerHeight};
  border-bottom: 2px solid transparent;

  ${(props) => props.isLink &&
    `&:hover,
    &:active,
    &:focus {
      ${focusStyle(props)}
    }`
  }
  ${(props) => props.checked &&
    focusStyle(props)
  }
`;

export default Item;
