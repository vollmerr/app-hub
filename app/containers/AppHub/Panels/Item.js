import React from 'react';
import styled from 'styled-components';

import Link from '../../../components/Link';
import theme from '../../../utils/theme';


export const Wrapper = styled.li`
  list-style-type: none;
  margin: 0 -20px;
  &:hover {
    background: ${theme.neutralLighter};
  }
`;


export const StlyedLink = styled(Link) `
  color: ${theme.white};
  display: block;
  padding: 15px;
  &:hover,
  &:active,
  &:focus {
    color: ${theme.neutralDark};
  }
`;


class Item extends React.PureComponent {
  render() {
    return (
      <Wrapper>
        <StlyedLink {...this.props} />
      </Wrapper>
    );
  }
}


export default Item;
