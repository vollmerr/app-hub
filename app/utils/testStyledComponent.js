import React from 'react';
import { shallow } from 'enzyme';
import 'jest-styled-components';

function testStyledComponent(Component, extend = null) {
  describe(`<${Component.displayName} />`, () => {
    let wrapper;
    beforeEach(() => {
      wrapper = shallow(<Component />);
    });

    it('should render correctly', () => {
      expect(wrapper).toMatchSnapshot();
    });

    it(`should render an <${Component.target}> tag`, () => {
      expect(wrapper.type()).toEqual(Component.target);
    });

    it('should have a className attribute', () => {
      expect(wrapper.prop('className')).toBeDefined();
    });

    it('should adopt a valid attribute', () => {
      const id = 'test';
      wrapper.setProps({ id });
      expect(wrapper.prop('id')).toEqual(id);
    });

    if (extend) {
      it(`should extend ${extend.name}`, () => {
        expect(wrapper.find(extend).exists()).toEqual(true);
      });
    } else {
      it('should not adopt an invalid attribute', () => {
        const attribute = 'test';
        wrapper.setProps({ attribute });
        expect(wrapper.prop('attribute')).toBeUndefined();
      });
    }
  });
}

export default testStyledComponent;

