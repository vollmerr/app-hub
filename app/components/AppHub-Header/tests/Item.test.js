import testStyledComponent from 'utils/testStyledComponent';

import Item from '../Item';

testStyledComponent(Item);
