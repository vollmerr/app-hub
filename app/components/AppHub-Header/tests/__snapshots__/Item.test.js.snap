// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Item__Item /> should render correctly 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  height: 50px;
}

<div
  className="c0"
/>
`;
